src/runtime_src/core/common/device_factory.h

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2023 Advanced Micro Devices, Inc. All rights reserved.
+#ifndef _XCL_DEV_FACTORY_H_
+#define _XCL_DEV_FACTORY_H_
+
+#include "device.h"
+#include <vector>
+
+namespace xrt_core {
+
+// One function on FPGA or AIE device
+class device_factory
+{
+public:
+  bool m_is_mgmt = false;
+  bool m_is_ready = true; 
+  device_factory(bool isuser) {  
+      m_is_mgmt = !isuser;  
+  }
```

Comment:
Leaving these bools public makes sense only if they can be `const` to prevent them from being changed by client code.  Anything else is bad design.  If `m_is_ready` cannot be initialized at construction time, then it cannot be a public data member.

Here is how to do is_mgmt (and do remove `m_`  for public data members).
```
  const bool is_mgmt {false};
  device_factory(bool isuser) 
    : m_is_msgmt{!isuser}
  {}
```
This prevents any modifications after construction.

Reply:
made both mgmt and ready variables private.